### ISSUE TITLE
[ENH]: Using with sklearn 1.2.0

### ISSUE THREAD

**Issue by ngupta23** [pycaret/pycaret#3199]
### Describe the feature you want to add to this project

Can not currently use pycaret with sklearn 1.2.0 due to deprecations in model parameters. 

Refer to this for more details: https://github.com/pycaret/pycaret/pull/3198#issue-1510107248

### Describe your proposed solution

Need to update the model parameters internally to make it work. But removing normalize may impact those that are using the older version of scikit learn (> 1.0, < 1.2)

### Describe alternatives you've considered, if relevant

Currently limiting the sklearn vesion to be < 1.2.0

**Comment by celestinoxp** [comment]
I think support of scikit-learn 1.2 should be a PRIORITY because scikit-learn is a base of pycaret.

More reasons for these urgence: Efficiency, Enhancements and a Fixes: 

https://scikit-learn.org/stable/whats_new/v1.2.html
